extensions/amp-graphiq/0.1/test/test-amp-graphiq.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createIframePromise,
+  doNotLoadExternalResourcesInTest,
+} from '../../../../testing/iframe';
+import '../amp-graphiq';
+import {adopt} from '../../../../src/runtime';
+import * as sinon from 'sinon';
+
+adopt(window);
+
+describe('amp-graphiq', () => {
+
+  let sandbox;
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  function getGraphiq(widgetId, opt_responsive, opt_beforeLayoutCallback) {
+    return createIframePromise(true, opt_beforeLayoutCallback).then(iframe => {
+      doNotLoadExternalResourcesInTest(iframe.win);
+      const graphiq = iframe.doc.createElement('amp-graphiq');
+      graphiq.setAttribute('data-widget-id', widgetId);
+      graphiq.setAttribute('width', '600');
+      graphiq.setAttribute('height', '512');
+      graphiq.setAttribute('alt', 'Testing');
+      if (opt_responsive) {
+        graphiq.setAttribute('layout', 'responsive');
+      }
+      // Placeholder
+      const img = iframe.doc.createElement('amp-img');
+      img.setAttribute('layout', 'fill');
+      img.setAttribute('src', 'https://i.ytimg.com/vi/cMcCTVAFBWM/hqdefault.jpg');
+      img.setAttribute('placeholder', '');
```

Comment:
Yup. You can define a [`#createPlaceholderCallback`](https://github.com/ampproject/amphtml/blob/a76e68b1ec648cbf71c9e0c8ecabecd4c0d40fbd/src/base-element.js#L350-L358) in this class.